pandas/core/construction.py

```diff
@@ -434,7 +434,9 @@ def sanitize_array(
             subarr = subarr.copy()
         return subarr
 
-    elif isinstance(data, (list, tuple)) and len(data) > 0:
+    elif isinstance(data, (list, tuple, set)) and len(data) > 0:
```

Comment:
Would it be appropriate to use `is_list_like(..., allow_sets=True)` here?

Reply:
Good question. It looks like that ends up interfering a bit with some of the other cases (`range` inputs that are handled differently below, as well `array.array` which causes a test to fail when I try to use that here).

Reply:
Alright sounds like it's not a good fit here (maybe make a comment here why we can't use `is_list_like`)